### ISSUE TITLE
there may be some bugs when calculate the loss in train or validation set

### ISSUE THREAD

**Issue by InTheFuture7** [YifanZhang-git/SKAPP#8]
**about the loss in train set, it just records the loss of the single batch with the min loss during the entire epoch, rather than the average loss across all batches?**

https://github.com/YifanZhang-git/SKAPP/blob/c35c7ebcd0612fa208bec457c12d5048570fbc09/src/RRCP/train_all_item.py#L146-L161

https://github.com/YifanZhang-git/SKAPP/blob/c35c7ebcd0612fa208bec457c12d5048570fbc09/src/RRCP/train_all_item.py#L123-L127

**about the loss in validation set, it is just sum all the batches's loss, not average them**

https://github.com/YifanZhang-git/SKAPP/blob/c35c7ebcd0612fa208bec457c12d5048570fbc09/src/RRCP/train_all_item.py#L167-L180
